server/controllers/finance/reports/collection_capacity/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @overview ./finance/reports/collection_capacity/
+ */
+
+const _ = require('lodash');
+const moment = require('moment');
+
+const ReportManager = require('../../../../lib/ReportManager');
+const db = require('../../../../lib/db');
+
+module.exports.report = report;
+
+// path to the template to render
+const TEMPLATE = './server/controllers/finance/reports/collection_capacity/report.handlebars';
+
+const DEFAULT_OPTIONS = {
+  orientation : 'portrait',
+  footerRight : '[page] / [toPage]',
+  footerFontSize : '7',
+};
+
+/**
+ * @method report
+ *
+ * @description
+ * The HTTP interface which actually creates the report.
+ */
+async function report(req, res, next) {
+  try {
+    const qs = _.extend(req.query, DEFAULT_OPTIONS);
+    const {
+      dateFrom, dateTo, accountId, cashboxId,
+    } = req.query;
+    const metadata = _.clone(req.session);
+
+    const rpt = new ReportManager(TEMPLATE, metadata, qs);
+
+    const CASH_PAYMENT_TRANSACTION_TYPE = 2;
+    const CAUTION_TRANSACTION_TYPE = 19;
+
+    const patients = `
+      SELECT DATE(p.registration_date) AS registration_date, COUNT(*) registrations
+      FROM patient p 
+      WHERE (DATE(p.registration_date) BETWEEN ? AND ?)
+        AND project_id = ?
+      GROUP BY DATE(p.registration_date)
+    `;
+
+    const invoices = `
+      SELECT DATE(i.date) AS invoice_date, SUM(i.cost) total_invoiced
+      FROM invoice i 
+      WHERE DATE(i.date) BETWEEN ? AND ?
+        AND reversed = 0 AND project_id = ?
+      GROUP BY DATE(i.date)
+    `;
+
+    const payments = `
+      SELECT DATE(gl.trans_date) trans_date, SUM(IFNULL(debit_equiv - credit_equiv, 0)) AS total_paid 
+      FROM general_ledger gl
+      JOIN cash c ON c.uuid = gl.record_uuid
+      WHERE gl.transaction_type_id IN (${CASH_PAYMENT_TRANSACTION_TYPE}, ${CAUTION_TRANSACTION_TYPE})
+        AND (DATE(gl.trans_date) >= ? AND DATE(gl.trans_date) <= ?)
```

Comment:
Hmm....  This is different from the invoices query.  On the invoices query, you are using the `invoice.date` as the basis for the calculation.  Here, instead of `cash.date` you are using the `general_ledger.trans_date`.  In my view, you should either:
  1. Use `cash.date`  OR
  2.  Use `gl.trans_date` for the invoices query.

Personally, I prefer the idea of using `gl.trans_date` for the invoices since someone could have edited the dates in the posting_journal.